packages/schematics/angular/utility/ast-utils.ts

```diff
@@ -642,27 +688,45 @@ export function addRouteDeclarationToModule(
       const { line } = source.getLineAndCharacterOfPosition(routesArg.getStart());
       throw new Error(
         `No route declaration array was found that corresponds ` +
-        `to router module at line ${line} in ${fileToAdd}`,
+          `to router module at line ${line} in ${fileToAdd}`,
       );
     }
 
-    routesArr = findNodes(routesVar, ts.SyntaxKind.ArrayLiteralExpression, 1)[0] as ts.ArrayLiteralExpression;
+    routesArr = findNodes(
+      routesVar,
+      ts.SyntaxKind.ArrayLiteralExpression,
+      1,
+    )[0] as ts.ArrayLiteralExpression;
   }
 
-  const occurencesCount = routesArr.elements.length;
+  const occurrencesCount = routesArr.elements.length;
   const text = routesArr.getFullText(source);
 
   let route: string = routeLiteral;
-  if (occurencesCount > 0) {
-    const identation = text.match(/\r?\n(\r?)\s*/) || [];
-    route = `,${identation[0] || ' '}${routeLiteral}`;
+  let insertPos = routesArr.elements.pos;
+
+  if (occurrencesCount > 0) {
+    const routeLiterals = findNodes(routesArr, ts.SyntaxKind.ObjectLiteralExpression);
```

Comment:
This whole part to determine if it's a wilecard route can be simplified a bit, and make it more safe and performant by removing the `getText` which is slower than using `.text` directly.

Something like the below should be enough.

```ts
    const lastRouteLiteral = [...routesArr.elements].pop();
    const lastRouteIsWildcard = ts.isObjectLiteralExpression(lastRouteLiteral) && lastRouteLiteral
      .properties
      .some(n => (
        ts.isPropertyAssignment(n)
        && ts.isIdentifier(n.name)
        && n.name.text === 'path'
        && ts.isStringLiteral(n.initializer)
        && n.initializer.text === '**'
      ));
```

Reply:
Thanks for the review, it does look better.